CMakeLists.txt

```diff
@@ -32,6 +35,10 @@ if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
 endif()
 
 option (PORTS_OF_CALL_BUILD_TESTING "Test the current installation")
+# off by default but possible to turn on
+cmake_dependent_option(PORTS_OF_CALL_TEST_USE_KOKKOS
+  "Use Kokkos offloading for tests"
+  OFF "PORTS_OF_CALL_BUILD_TESTING" OFF)
```

Comment:
What about `PORTABILITY_STRATEGY_CUDA`?

Reply:
I would expect something like `TEST_PORTABILITY_STRATEGY` that can be None, Cuda, or Kokkos.

Reply:
```cmake
set(PORTS_OF_CALL_TEST_PORTABILITY_STRATEGY "None" CACHE STRING "Portability strategy used by tests")
set_property(CACHE PORTS_OF_CALL_TEST_PORTABILITY_STRATEGY PROPERTY STRINGS None Cuda Kokkos)
```

Reply:
And instead of using `cmake_dependent_option` just use the existing `if(PORTS_OF_CALL_BUILD_TESTING)` below.

Reply:
:+1: done. And added kokkos fetching with fetch content if needed too. I'm making the cuda build throw a fatal error for now though as I haven't tested it carefully.